modules/reference/pages/data-transforms/js/js-sdk.adoc

```diff
@@ -0,0 +1,137 @@
+= JavaScript API for Data Transforms
+:description: Work with data transforms using JavaScript.
+:page-aliases: reference:data-transforms/js-sdk.adoc
+
+This page contains the API reference for the data transforms client library of the JavaScript SDK.
+
+== Functions
+
+[[OnRecordWritten]]
+=== onRecordWritten()
+
+  onRecordWritten (`cb`): `void`
+
+Registers a callback to be fired when a record is written to the input topic. This callback is triggered after the record has been written, fsynced to disk, and acknowledged by the producer. This method should be called in your script's entry point.
+
+==== Parameters
+
+- <<OnRecordWrittenCallback, `cb`>>
+
+==== Returns
+
+`void`
+
+==== Example
+
+```ts
+import {onRecordWritten} from "@redpanda-data/transform-sdk";
+
+// Copy the input data to the output topic.
+onRecordWritten((event, writer) => {
+  writer.write(event.record);
+});
+```
+
+== Interfaces
+
+[[OnRecordWrittenCallback]]
+=== OnRecordWrittenCallback()
+
+  OnRecordWrittenCallback : (`event`, `writer`) => `void`
+
+The callback type for <<OnRecordWritten, `OnRecordWritten`>>.
+
+==== Parameters
+
```

Comment:
For consistency, should this be Properties instead of Parameters?

Reply:
When referring to functions, the correct term is parameters. Properties are available in objects (interfaces).